server/bundler/extensions-loader.js

```diff
@@ -0,0 +1,74 @@
+const fs = require( 'fs' );
+const path = require( 'path' );
+const camelCase = require( 'lodash' ).camelCase;
+const compact = require( 'lodash' ).compact;
+
+function getReducerPath( extensionDir ) {
+	return path.join( 'client', 'extensions', extensionDir, 'state', 'reducer.js' );
+}
+
+function generateReducerRequireString( extensionDir ) {
+	return `'${ camelCase( extensionDir ) }': require( 'extensions/${ extensionDir }/state/reducer' )\n`;
+}
+
+function generateExtensionsModuleString( reducerRequires ) {
+	return `module.exports = {
```

Comment:
Could it have the following code instead?

```js
return `
const combineReducers = require( ''redux' ).combineReducers;

module.exports = combineReducers(
    ${ reducerRequires }
);`;

Then it `client/state/index.js` it would be enough to have the following import statement:
```js
import extensions from 'extensions';
```